app/controllers/api/v1/workflows_controller.rb

```diff
@@ -34,14 +34,11 @@ def index
       end
 
       def destroy
-        workflow = Workflow.find(params.require(:id))
-        workflow.destroy!
+        WorkflowDeleteService.new(params.require(:id)).delete
         head :no_content
       rescue ActiveRecord::InvalidForeignKey => e
         json_response({ :message => e.message }, :forbidden)
-      rescue ActiveRecord::RecordNotDestroyed
-        raise unless workflow.errors[:base].include?(Workflow::MSG_PROTECTED_RECORD)
-
+      rescue Exceptions::NotAuthorizedError
```

Comment:
why to remove `RecordNotDestoryed` and why specially for `NotAuthorizedError`? Shouldn't it be commonly handled for all controller methods?